site/content/in-dev/unreleased/configuring-polaris-for-production.md

```diff
@@ -23,114 +23,180 @@ type: docs
+By default, Polaris uses `rsa-key-pair`, with randomly generated keys. 
 
-## Metastore Management
+> [!IMPORTANT]
+> The default `rsa-key-pair` configuration is not suitable when deploying many replicas of Polaris,
+> as each replica will have its own set of keys. This will cause token validation to fail when a
+> request is routed to a different replica than the one that issued the token.
 
-> [!IMPORTANT]  
-> The default `in-memory` implementation for `metastoreManager` is meant for testing and not suitable for production usage. Instead, consider an implementation such as `eclipse-link` which allows you to store metadata in a remote database.
+It is highly recommended to configure Polaris with previously-generated RSA keys. This can be done
+by setting the following properties:
 
-A Metastore Manger should be configured with an implementation that durably persists Polaris entities. Use the configuration `metaStoreManager` to configure a [MetastoreManager](https://github.com/apache/polaris/blob/627dc602eb15a3258dcc32babf8def34cf6de0e9/polaris-core/src/main/java/io/polaris/core/persistence/PolarisMetaStoreManager.java#L47) implementation where Polaris entities will be persisted. 
+```properties
+polaris.authentication.token-broker.type=rsa-key-pair
+polaris.authentication.token-broker.rsa-key-pair.public-key-file=/tmp/public.key
+polaris.authentication.token-broker.rsa-key-pair.private-key-file=/tmp/private.key
+```
 
-Be sure to secure your metastore backend since it will be storing credentials and catalog metadata.
+Alternatively, you can use a symmetric key by setting the following properties:
 
-### Configuring EclipseLink
+```properties
+polaris.authentication.token-broker.type=symmetric-key
+polaris.authentication.token-broker.symmetric-key.file=/tmp/symmetric.key
+```
 
-To use EclipseLink for metastore management, specify the configuration `metaStoreManager.conf-file` to point to an EclipseLink `persistence.xml` file. This file, local to the Polaris service, contains details of the database used for metastore management and the connection settings. For more information, refer to the [metastore documentation]({{% ref "metastores" %}}).
+Note: it is also possible to set the symmetric key secret directly in the configuration file, but
+that is not recommended for production use, as the secret is stored in plain text:
 
-> [!IMPORTANT]
-> EclipseLink requires
-> 1. Building the JAR for the EclipseLink extension
-> 2. Setting the `eclipseLink` gradle property to `true`.
->
-> This can be achieved by setting `eclipseLink=true` in the `gradle.properties` file, or by passing the property explicitly while building all JARs, e.g.: `./gradlew -PeclipseLink=true clean assemble`
+```properties
+polaris.authentication.token-broker.symmetric-key.secret=my-secret
```

Comment:
Is it possible to get the secrets from env variables?

Reply:
Yes that's possible.